### PR TITLE
All  jobs page

### PR DESCRIPTION
**Pull request by Mohit-Sony**

### CODE REVIEW

**Comment by Mohit-Sony** [comment]
search btn and adding updated

File: package-lock.json
Comment (panthsolanki) [inline package-lock.json:1]: pls remove this file

File: pages/careers/AllJobs.vue [hunk 1]
```diff
@@ -0,0 +1,17 @@
+<template>
+    <NuxtLayout name="zen">
+        <AllJobsPage/>
+    </NuxtLayout>
+</template>
+
+<script>
+// import '@/assets/css/tailwind/allJobs.css'
```
Comment (panthsolanki) [inline pages/careers/AllJobs.vue]: Remove unwanted code

File: components/Careers/SingleJobCard.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    export default {
+        name:'SingleJobCard',
+        props:{
+            data:Object
+        }
+    }
+</script>
+
+<style scoped>
+
```
Comment (panthsolanki) [inline components/Careers/SingleJobCard.vue]: remove style if not needed

**Comment by Mohit-Sony** [comment]
all bugs fixed

File: components/Careers/AllJobsPage.vue [hunk 3]
```diff
@@ -0,0 +1,427 @@
+<template>
+    <body class="font-nunito text-base text-black dark:text-white dark:bg-slate-900">
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: Update component name: Page should not be used in component name

File: components/Careers/AllJobsPage.vue [hunk 4]
```diff
@@ -0,0 +1,427 @@
...
+            <div class="container">
+                <div class="grid lg:grid-cols-12 grid-cols-1" id="reserve-form">
+                    <div class="lg:col-start-2 lg:col-span-10">
+                        <div class="bg-white dark:bg-slate-900 border-0 shadow dark:shadow-gray-800 rounded p-3 -mt-[150px]">
+                            <form action="#">
+                                <div class="registration-form text-dark text-start">
+                                    <div class="grid lg:grid-cols-4 md:grid-cols-2 grid-cols-1 lg:gap-0 gap-6">
+                                        <div class="filter-search-form relative filter-border">
+                                            <i class="uil uil-briefcase-alt icons"></i>
+                                            <input v-model="currentdata.keyword" name="name" type="text" id="job-keyword" class="form-input filter-input-box bg-gray-50 dark:bg-slate-800 border-0" placeholder="Search your keywords">
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: do code formatting

File: components/Careers/AllJobsPage.vue (hunk 3 above)
Comment (panthsolanki) [reply components/Careers/AllJobsPage.vue]: you can directly put all this code in page folder, why to make component like this ?

File: components/Careers/AllJobsPage.vue [hunk 5]
```diff
@@ -0,0 +1,427 @@
...
+    
+            </div>
+            <!--end container-->
+    
+            <div class="grid md:grid-cols-12 grid-cols-1 mt-8">
+                <div class="md:col-span-12 text-center">
+                    <nav aria-label="Page navigation example">
+                        <ul class="inline-flex items-center -space-x-px">
+                            <li v-if="currentPage != 1" @click="currentPage--">
+                                <a href="#" class="w-[40px] h-[40px] inline-flex justify-center items-center text-slate-400 bg-white dark:bg-slate-900 ltr:rounded-l-lg rtl:rounded-r-lg hover:text-white border border-gray-100 dark:border-gray-700 hover:border-indigo-600 dark:hover:border-indigo-600 hover:bg-indigo-600 dark:hover:bg-indigo-600">
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: do code formatting

File: components/Careers/AllJobsPage.vue [hunk 6]
```diff
@@ -0,0 +1,427 @@
...
+                currentPage: 1,
+                perPage: 6,
+            }
+        },
+        computed: {
+            sortedLocations() {
+                // Extract all locations from jobs data
+                const locations = this.jobsdata.map((job) => job.location);
+    
+                // Remove duplicates and sort the locations
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: remove unwanted comment

File: components/Careers/AllJobsPage.vue [hunk 7]
```diff
@@ -0,0 +1,427 @@
...
+    
+        },
+        methods: {
+            getPaginatedData() {
+                const start = (this.currentPage - 1) * this.perPage;
+                const end = start + this.perPage;
+                const paginated = this.filteredJobs.slice(start, end);
+                return paginated;
+            },
+            subitHandler(){
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: fix the typo, it should be submitHandler
also its not submit but search button clicked 
please give name according to its functionality

File: assets/css/tailwind/choices.min.css [hunk 8]
```diff
@@ -0,0 +1,6 @@
+.choices{
```
Comment (panthsolanki) [inline assets/css/tailwind/choices.min.css:1]: if this file is not used, please remove

File: components/Careers/AllJobsPage.vue [hunk 9]
```diff
@@ -0,0 +1,427 @@
...
+                <!--grid-->
+            </div>
+            <!--end container-->
+            <!-- form container ends -->
+
+    
+            <div class="container">
+                <div class="grid lg:grid-cols-3 md:grid-cols-2 grid-cols-1 mt-8 gap-[30px]">
+    
+                    <SingleJobCard v-for="job in getPaginatedData()" :data="job" />
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: if there is no job for particular search, then show the message "There are currently no job openings that match the selected filter."

File: components/Careers/AllJobsPage.vue [hunk 10]
```diff
@@ -0,0 +1,427 @@
...
+    
+                </div>
+                <!--end grid-->
+    
+            </div>
+            <!--end container-->
+    
+            <div class="grid md:grid-cols-12 grid-cols-1 mt-8">
+                <div class="md:col-span-12 text-center">
+                    <nav aria-label="Page navigation example">
```
Comment (panthsolanki) [inline components/Careers/AllJobsPage.vue]: if there is no job for particular search, then remove the pagination section

File: assets/css/tailwind/choices.min.css
Comment (panthsolanki) [inline assets/css/tailwind/choices.min.css:1]: if this file is not used, please remove

File: package-lock.json
Comment (Mohit-Sony) [reply package-lock.json:1]: remove means git ignore or delete

File: components/Careers/AllJobsPage.vue (hunk 3 above)
Comment (Mohit-Sony) [reply components/Careers/AllJobsPage.vue]: I had to scope the css for this component only.

File: components/Careers/AllJobsPage.vue (hunk 9 above)
Comment (Mohit-Sony) [reply components/Careers/AllJobsPage.vue]: done

File: components/Careers/AllJobsPage.vue (hunk 10 above)
Comment (Mohit-Sony) [reply components/Careers/AllJobsPage.vue]: done

File: assets/css/tailwind/choices.min.css
Comment (Mohit-Sony) [reply assets/css/tailwind/choices.min.css:1]: done

File: components/Careers/AllJobsPage.vue (hunk 4 above)
Comment (Mohit-Sony) [reply components/Careers/AllJobsPage.vue]: have a look

File: components/Careers/AllJobsPage.vue (hunk 5 above)
Comment (Mohit-Sony) [reply components/Careers/AllJobsPage.vue]: have a look

**Comment by Mohit-Sony** [comment]
issues resolved in commit [pr bugs fixed v1](https://github.com/pewebdevelopment/ZenVerse/pull/9/commits/3ea46615dc2ed4a839bbe583a3d32aa562b1d52f)

**Pull request merged**